src/pages/createplan/[id].tsx

```diff
@@ -271,16 +290,28 @@ const CreatePlan = ({
   return (
     <div className="flex min-h-screen flex-col overflow-x-hidden">
       {/* <Navbar /> */}
-      <div className="flex max-h-20 min-h-20 items-center gap-4 bg-mainbutton5">
-        <div className="flex-none">
-          <Logo />
+      <div className="flex max-h-20 min-h-20 items-center justify-between bg-mainbutton5">
+        <div className="ml-4 w-1/4 flex-none">
+          <SolvroLogo />
         </div>
-        <div className="flex flex-grow justify-center">
-          <h1 className="flex cursor-pointer items-center gap-2 rounded border border-dashed p-2 text-center text-sm font-semibold text-white md:px-14 md:text-3xl">
-            Mój plan <CiEdit />
-          </h1>
+        <div className="flex items-center justify-center rounded-xl bg-mainbutton2 p-1.5 text-lg font-semibold md:text-3xl">
+          <input
+            type="text"
+            value={plan.name}
+            onChange={(e) => {
+              changePlanName(e.target.value);
+            }}
+            className={`w-full truncate border-mainbutton5 bg-transparent outline-none duration-100 ease-in-out before:transition-all ${
+              isEditing ? "border-b-2 font-normal" : " "
+            }`}
+            disabled={!isEditing}
+          />
+          <CiEdit
+            className="ml-2 transform cursor-pointer transition-transform hover:scale-110 active:scale-90"
+            onClick={handleEditClick}
+          />
```

Comment:
i też tbh, przy edytowaniu fajnie jakby to się na jakiś checkmark zamieniało?

w sensie
nie da się edytować -> przycisk ten CiEdit
da się edytować -> jakiś checkmark co oznacza zapisanie

i też tbh, zrobiłbym to jako button z shadcn'a

https://ui.shadcn.com/docs/components/button#icon